ducktrack/elements.py

```diff
@@ -755,10 +765,17 @@ def track(self,p):
             p.delta += self.gauss_noise_ampl_s*np.random.randn(len(p.delta))
 
         # re-adding dispersion and closed orbit
+        old_x=p.x
+        old_px=p.px
+        old_y=p.y
+        old_py=p.py
+
```

Comment:
This is not a copy but just a reference.

To have a full copy of the array you need:
```python
old_x = p.x.copy()
```

Reply:
Fixed